packages/manager/cypress/e2e/core/cloudpulse/timerange-verification.spec.ts

```diff
@@ -0,0 +1,473 @@
+      supportedRegionIds: '',
+    },
+    {
+      dimensionKey: 'cluster_id',
+      maxResourceSelections: 10,
+      serviceType: 'dbaas',
+      supportedRegionIds: 'us-ord',
+    },
+  ],
+};
+
+const {
+  metrics,
+  id,
+  serviceType,
+  dashboardName,
+  engine,
+  nodeType,
+} = widgetDetails.dbaas;
+
+const dashboard = dashboardFactory.build({
+  label: dashboardName,
+  service_type: serviceType,
+  widgets: metrics.map(({ title, yLabel, name, unit }) => {
+    return widgetFactory.build({
+      label: title,
+      y_label: yLabel,
+      metric: name,
+      unit,
+    });
+  }),
+});
+
+const metricDefinitions = {
+  data: metrics.map(({ title, name, unit }) =>
+    dashboardMetricFactory.build({
+      label: title,
+      metric: name,
+      unit,
+    })
+  ),
+};
+
+const mockAccount = accountFactory.build();
+
+const metricsAPIResponsePayload = cloudPulseMetricsResponseFactory.build({
+  data: generateRandomMetricsData('Last 30 Days', '1 day'),
+});
+
+const databaseMock: Database = databaseFactory.build({
+  type: engine,
+  region: mockRegion.label,
+});
+
+const timezone = 'Asia/Kolkata';
+
+/**
+ * Generates a date in Indian Standard Time (IST) based on a specified number of days offset,
+ * hour, and minute. The function also provides individual date components such as day, hour,
+ * minute, month, and AM/PM.
+ *
+ * @param {number} daysOffset - The number of days to adjust from the current date. Positive
+ *                               values give a future date, negative values give a past date.
+ * @param {number} hour - The hour to set for the resulting date (0-23).
+ * @param {number} [minute=0] - The minute to set for the resulting date (0-59). Defaults to 0.
+ *
+ * @returns {Object} - Returns an object containing:
+ *   - `actualDate`: The formatted date and time in IST (YYYY-MM-DD HH:mm).
+ *   - `day`: The day of the month as a number.
+ *   - `hour`: The hour in the 24-hour format as a number.
+ *   - `minute`: The minute of the hour as a number.
+ *   - `month`: The month of the year as a number.
+ *   - `ampm`: The AM/PM designation of the time (either 'AM' or 'PM').
+ */
+const getDateRangeInIST = (
+  daysOffset: number,
+  hour: number,
+  minute: number = 0
+) => {
+  const now = DateTime.now().setZone(timezone);
+  const targetDate = now
+    .startOf('month')
+    .plus({ days: daysOffset })
+    .set({ hour, minute });
+
+  const actualDate = targetDate.toFormat('yyyy-LL-dd HH:mm');
+  return {
+    actualDate,
+    day: targetDate.day,
+    hour: targetDate.hour,
+    minute: targetDate.minute,
+    month: targetDate.month,
+  };
+};
+
+/**
+ * This function calculates the start and end of the previous month,
+ * adjusted by subtracting 5 hours and 30 minutes, and returns them
+ * in the ISO 8601 format (UTC).
+ *
+ * @returns {{start: string, end: string}} - The start and end dates of the previous month in ISO 8601 format.
+ */
+const getLastMonthRange = (): { start: string; end: string } => {
```

Comment:
https://github.com/ACLPManager/manager/blob/8976a572f7425d3b7c4adb2593a84befa945be49/packages/api-v4/src/cloudpulse/types.ts#L51

set return type to this type
```suggestion
const getLastMonthRange = (): DateTimeWIthPreset => {
```